src/sage/graphs/homomorphisms/count_homomorphisms.py

```diff
@@ -0,0 +1,236 @@
+from sage.graphs.graph import Graph
+
+from sage.graphs.homomorphisms.helper_functions import *
+
+from sage.graphs.graph_decompositions.tree_decomposition import make_nice_tree_decomposition, label_nice_tree_decomposition
```

Comment:
these imports are used only once in `__init__` and so can be done just before using them.